projects/igniteui-angular/src/lib/directives/input/input.directive.spec.ts

```diff
@@ -663,6 +669,78 @@ describe('IgxInput', () => {
         igxInput.value = 'Test';
         expect(igxInput.value).toBe('Test');
     });
+
+    it('Should properly initialize when used as a reactive form control - without initial validators/toggle validators', fakeAsync(() => {
+        const fix = TestBed.createComponent(InputReactiveFormComponent);
+        fix.detectChanges();
+        // 1) check if label's --required class and its asterisk are applied
+        const dom = fix.debugElement;
+        const input = fix.componentInstance.input;
+        const formGroup: FormGroup = fix.componentInstance.reactiveForm;
+        let inputGroupIsRequiredClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
+        let inputGroupInvalidClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_INVALID));
+        // interaction test - expect actual asterisk
+        // The only way to get a pseudo elements like :before OR :after is to use getComputedStyle(element [, pseudoElt]),
+        // as these are not in the actual DOM
+        let asterisk = window.getComputedStyle(dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_LABEL)).nativeElement, ':after').content;
+        expect(asterisk).toBe('"*"');
+        expect(inputGroupIsRequiredClass).toBeDefined();
+        expect(inputGroupIsRequiredClass).not.toBeNull();
+
+        // 2) check that input group's --invalid class is NOT applied
+        expect(inputGroupInvalidClass).toBeNull();
+
+        // interaction test - markAsTouched + focus&blur, so the --invalid and --required classes are applied
+        fix.debugElement.componentInstance.markAsTouched();
+        const inputGroup = fix.debugElement.query(By.css('.' + INPUT_GROUP_CSS_CLASS));
+        inputGroup.nativeElement.click();
```

Comment:
Fairly certain you can do focus+blur by.. just triggering the handlers. Also blur's job is kind of to set touched, so you shouldn't need to to set it in advance.

Reply:
However, still need to set  set touched as the component does not